### ISSUE TITLE
kong can't start by `xpcall` 

### ISSUE THREAD

**Issue by NICK-DUAN** [Kong/kong#12418]
In `kong/charts`, I change the args with `["kong", "docker-start", "--vv"]` for proxy container, And I get this log:
```bash
Error: 
/usr/local/share/lua/5.1/kong/cmd/utils/inject_confs.lua:28: 
stack traceback:
	[C]: in function 'assert'
	/usr/local/share/lua/5.1/kong/cmd/utils/inject_confs.lua:28: in function 'load_conf'
	/usr/local/share/lua/5.1/kong/cmd/utils/inject_confs.lua:85: in function </usr/local/share/lua/5.1/kong/cmd/utils/inject_confs.lua:84>
	[C]: in function 'xpcall'
	/usr/local/bin/kong:99: in function 'file_gen'
	init_worker_by_lua(nginx.conf:189):51: in function <init_worker_by_lua(nginx.conf:189):49>
	[C]: in function 'xpcall'
	init_worker_by_lua(nginx.conf:189):58: in function <init_worker_by_lua(nginx.conf:189):56>
```

I don't know why, maybe in this project someone can help me?

**Comment by NICK-DUAN** [comment]
but i run with docker it's ok:
```bash
docker run -it kong:3.5 bash    
# Enter the container
export KONG_DATABASE=off
./docker-entrypoint.sh kong docker-start --vv
```
get normal log
```bash
2024/01/25 03:24:27 [notice] 61#0: start worker process 1343
2024/01/25 03:24:27 [notice] 61#0: start worker process 1344
2024/01/25 03:24:27 [notice] 61#0: start worker process 1345
2024/01/25 03:24:27 [notice] 61#0: start worker process 1346
2024/01/25 03:24:27 [notice] 61#0: start worker process 1347
2024/01/25 03:24:27 [notice] 61#0: start worker process 1348
2024/01/25 03:24:27 [notice] 1333#0: *3 [lua] globalpatches.lua:73: sleep(): executing a blocking 'sleep' (0.001 seconds), context: init_worker_by_lua*
2024/01/25 03:24:27 [notice] 1339#0: *1 [lua] globalpatches.lua:73: sleep(): executing a blocking 'sleep' (0.001 seconds), context: init_worker_by_lua*
2024/01/25 03:24:27 [notice] 1339#0: *1 [lua] globalpatches.lua:73: sleep(): executing a blocking 'sleep' (0.002 seconds), context: init_worker_by_lua*
2024/01/25 03:24:27 [notice] 1333#0: *3 [lua] globalpatches.lua:73: sleep(): executing a blocking 'sleep' (0.002 seconds), context: init_worker_by_lua*
2024/01/25 03:24:27 [notice] 1339#0: *1 [lua] globalpatches.lua:73: sleep(): executing a blocking 'sleep' (0.004 seconds), context: init_worker_by_lua*
2024/01/25 03:24:27 [notice] 1333#0: *3 [lua] globalpatches.lua:73: sleep(): executing a blocking 'sleep' (0.004 seconds), context: init_worker_by_lua*
2024/01/25 03:24:27 [notice] 1334#0: *2 [lua] init.lua:259: purge(): [DB cache] purging (local) cache, context: init_worker_by_lua*
2024/01/25 03:24:27 [notice] 1334#0: *2 [lua] init.lua:259: purge(): [DB cache] purging (local) cache, context: init_worker_by_lua*
2024/01/25 03:24:27 [notice] 1334#0: *2 [lua] globalpatches.lua:73: sleep(): executing a blocking 'sleep' (0.001 seconds), context: init_worker_by_lua*
2024/01/25 03:24:27 [notice] 1343#0: *11 [lua] globalpatches.lua:73: sleep(): executing a blocking 'sleep' (0.001 seconds), context: init_worker_by_lua*
```

get progresses:
```bash
kong@6441af79ffa6:/$ ps aux 
USER         PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
kong           1  0.0  0.0   4604  3744 pts/0    Ss   03:23   0:00 bash
kong          61  1.3  0.1 420884 63688 pts/0    T    03:24   0:00 nginx: master process /usr/local/openresty/nginx/sbin/nginx -p /usr/local/kong -c n
kong        1333  0.2  0.3 2560824 98896 pts/0   T    03:24   0:00 nginx: worker process
kong        1334  0.2  0.3 2560560 98716 pts/0   T    03:24   0:00 nginx: worker process
kong        1335  0.2  0.3 2560496 98252 pts/0   T    03:24   0:00 nginx: worker process
kong        1336  0.2  0.3 2560560 98340 pts/0   T    03:24   0:00 nginx: worker process
kong        1337  0.2  0.3 2560560 98472 pts/0   T    03:24   0:00 nginx: worker process
kong        1338  0.2  0.3 2560496 98620 pts/0   T    03:24   0:00 nginx: worker process
kong        1339  0.2  0.3 2560496 98720 pts/0   T    03:24   0:00 nginx: worker process
kong        1340  0.2  0.3 2560496 98600 pts/0   T    03:24   0:00 nginx: worker process
kong        1341  0.2  0.3 2560496 98384 pts/0   T    03:24   0:00 nginx: worker process
kong        1342  0.2  0.3 2560496 98396 pts/0   T    03:24   0:00 nginx: worker process
kong        1343  0.2  0.3 2560560 98764 pts/0   T    03:24   0:00 nginx: worker process
kong        1344  0.2  0.3 2560496 98428 pts/0   T    03:24   0:00 nginx: worker process
kong        1345  0.2  0.3 2560496 98352 pts/0   T    03:24   0:00 nginx: worker process
kong        1346  0.2  0.3 2560496 98608 pts/0   T    03:24   0:00 nginx: worker process
kong        1347  0.2  0.3 2560560 98400 pts/0   T    03:24   0:00 nginx: worker process
kong        1348  0.2  0.3 2560496 98472 pts/0   T    03:24   0:00 nginx: worker process
kong        1445  0.0  0.0   7056  1556 pts/0    R+   03:25   0:00 ps aux
```

**Comment by Water-Melon** [comment]
https://github.com/Kong/kong/blob/master/kong/cmd/utils/inject_confs.lua#L28
```diff
- assert(prepare_prefix(conf, nil, true))
+ local _, err = assert(prepare_prefix(conf, nil, true))
+ print(err)

return conf
```
You can fetch the return values of assert and print the second one to see what the error message is.

**Comment by NICK-DUAN** [comment]
> ```diff
> + local _, err = assert(prepare_prefix(conf, nil, true))
> ```

This may be caused by the running permissions, maybe I found what the problem exactly is, So close this issue, and trace in this issue: https://github.com/Kong/charts/issues/526#issuecomment-1914348781

**Comment by dataops-sre** [comment]
Hi, I run into exact same problem recently, I try to spin up kubernetes-dashboard(it uses kong dbless as gateway) in my cluster, it works fine in the local kind cluster. In the remote cluster it failed on kong dbless component. After some research, I have added `kong docker-start --vv` in kong deployment, I got exactly the same `xpcall` issue. 
It turned out the issue is due to the underline kubernetes node OS, we run our managed kubernetes cluster in OpenTelekom cloud, kubernetes workers run in CentOS 7.7 OS, by changing worker node OS fixed our issue and `xpcal` issue disapears in kong deployment. I hope it helps.
